src/test/java/com/puppycrawl/tools/checkstyle/api/DetailASTTest.java

```diff
@@ -143,15 +143,30 @@ private static void checkFile(String filename) throws Exception {
         final FileContents contents = new FileContents(text);
         final DetailAST rootAST = TreeWalker.parse(contents);
         if (rootAST != null) {
-            checkTree(rootAST, null, null, filename, rootAST);
+            checkTree(filename, rootAST);
         }
     }
 
-    private static void checkTree(final DetailAST node,
-                           final DetailAST parent,
-                           final DetailAST prev,
-                           final String filename,
-                           final DetailAST root) {
+    private static void checkTree(final String filename, final DetailAST root) {
+        DetailAST curNode = root;
+        while (curNode != null) {
+            checkNode(curNode, curNode.getParent(), curNode.getPreviousSibling(), filename, root);
```

Comment:
Information sent to the method can't be using the `DetailAST` methods.
The whole point of this test is to verify those exact methods are functioning properly.

Example: Parameter `prev` is set to `curNode.getPreviousSibling()` and inside the method is verified by `assertEquals(badPrevMsg, prev, node.getPreviousSibling());`.
All it is doing now is testing the method against itself. If you change `getPreviousSibling()` to return an invalid value, test would still pass.

Original test kept track of the parent and previous itself. This is why.